contrib/completion/bash/docker

```diff
@@ -5486,7 +5486,7 @@ _docker_wait() {
 	_docker_container_wait
 }
 
-COMPOSE_PLUGIN_PATH=$(docker info --format '{{json .ClientInfo.Plugins}}' | sed -n 's/.*"Path":"\([^"]\+docker-compose\)".*/\1/p')
+COMPOSE_PLUGIN_PATH=$(docker info --format '{{json .ClientInfo.Plugins}}' | sed -n -E 's/.*"Path":"([^"]+docker-compose)".*/\1/p')
```

Comment:
I was wondering if we could do it with `--format` alone, and (while the syntax is a bit awkward), looks like it's possible when combining `range` and `if eq`;

```bash
docker info --format '{{range .ClientInfo.Plugins}}{{if eq .Name "compose"}}{{.Path}}{{end}}{{end}}'
/usr/local/lib/docker/cli-plugins/docker-compose
```

The script would then look like;

```bash
COMPOSE_PLUGIN_PATH=$(docker info --format '{{range .ClientInfo.Plugins}}{{if eq .Name "compose"}}{{.Path}}{{end}}{{end}}')
```

Reply:
There's one small difference; the `docker info --format` always prints a newline, whereas the `sed` variant does not; if I replace `compose` with `nosuch` (to mimic a situation where the compose plugin is not installed);

```bash
docker info --format '{{range .ClientInfo.Plugins}}{{if eq .Name "nosuch"}}{{.Path}}{{end}}{{end}}'
# prints an empty line

docker info --format '{{json .ClientInfo.Plugins}}' | sed -n -E 's/.*"Path":"([^"]+docker-nosuch)".*/\1/p'
# doesn't print an empty line
```

But I don't think that's a problem as the script checks if `$COMPOSE_PLUGIN_PATH` contains a valid path, and that still looks to work ok with that;


```bash
# for the "nosuch" plugin:

COMPOSE_PLUGIN_PATH=$(docker info --format '{{range .ClientInfo.Plugins}}{{if eq .Name "nosuch"}}{{.Path}}{{end}}{{end}}')
if [ -f "$COMPOSE_PLUGIN_PATH" ] ; then echo YES; fi

# doesn't print anything

# for the "compose" plugin:
COMPOSE_PLUGIN_PATH=$(docker info --format '{{range .ClientInfo.Plugins}}{{if eq .Name "compose"}}{{.Path}}{{end}}{{end}}')
if [ -f "$COMPOSE_PLUGIN_PATH" ] ; then echo YES; fi

# prints "YES"
```

Reply:
@thaJeztah Your suggestion has the advantage that it accesses the information in a structured way (i.e. by `.Name`) instead of text based operations that rely on naming conventions.
So I think we should go this way.

Reply:
@albers Should do I bring a new PR and decline this one with the solution suggested by @thaJeztah or just force push a new commit and change the PR description and the name? 
I'm asking because the PR already has approves and assigned milestone. 
As a compromise, I could bring suggestions by @thaJeztah in a new PR for the next milestone (not the assigned one - 21.xx)

Reply:
I think it's best to force-push and then update the PR title to a more intention revealing wording, something like "make compose plugin detection in bash completion work on Mac OS"

Reply:
@albers It's done